### ISSUE TITLE
TSLint `exclude` does not work when used with the Node API

### ISSUE THREAD

**Issue by saadq** [palantir/tslint#3881]
### How to reproduce
I have created a minimal repo that demonstrates the problem [here](https://github.com/saadq/misc/tree/master/tslint-ignore). 
The `readme.md` gives the exact steps on how to reproduce the issue. 
TLDR: `npm run cli` uses the `exclude` option while `npm run api` does not.

### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.8.3
- __Running TSLint via__: Node.js API

### Details

**tslint.json**:

```json
{
  "defaultSeverity": "error",
  "extends": [
    "tslint:recommended"
  ],
  "linterOptions": {
    "exclude": [
      "src/files-to-lint/ignored-file.ts"
    ]
  }
}
```

**File structure**:

```
.
├── src
│   ├── api.ts
│   └── files-to-lint
│       ├── ignored-file.ts
│       └── linted-file.ts
├── tsconfig.json
└── tslint.json
```

**ignored-file.ts**:

```ts
// This should be ignored
const x: number = 10
```

**linted-file.ts**:

```ts
// This should be linted
const x: number = 10
```

**api.ts**:

```ts
import { Configuration, Linter } from "tslint";

const project = ".";
const format = "stylish";
const configFile = "tslint.json";

const options = {
  fix: false,
  formatter: format,
};

const program = Linter.createProgram("tsconfig.json", project);
const linter = new Linter(options, program);
const files = Linter.getFileNames(program);

files.forEach((file) => {
  const fileContents = program.getSourceFile(file)!.getFullText();
  const config = Configuration.findConfiguration(configFile, file).results;
  linter.lint(file, fileContents, config);
});

const results = linter.getResult();
process.stdout.write(results.output);
process.exit(1);
```

#### Actual behavior

```
/Users/squadri/Projects/misc/tslint-ignore/src/files-to-lint/ignored-file.ts:2:21
ERROR: 2:21  semicolon  Missing semicolon

/Users/squadri/Projects/misc/tslint-ignore/src/files-to-lint/linted-file.ts:2:21
ERROR: 2:21  semicolon  Missing semicolon
```

#### Expected behavior

```
/Users/squadri/Projects/misc/tslint-ignore/src/files-to-lint/linted-file.ts:2:21
ERROR: 2:21  semicolon  Missing semicolon
```

**Comment by giladgray** [comment]
this kinda seems expected given that it's in the `linterOptions` block, which you define at the top of your script `const options` long before the config file has actually been loaded. this is likely an edge case in the design of the Node API where that field isn't properly supported due to the lifecycle of the components in Node. 

try setting `exclude` in your options object above. you could actually `require("path/to/tslint.json")` instead of using `findConfiguration` so you can refer to `linterOptions.exclude` instead of copy-paste.

**Comment by saadq** [comment]
I tried putting the `exclude` field in the `options` object, but it doesn't seem to work. Additionally, the declaration file for the `options` object you pass to the `Linter` constructor seems to look like this:

```
export interface ILinterOptions {
    fix: boolean;
    formatter?: string | FormatterConstructor;
    formattersDirectory?: string;
    rulesDirectory?: string | string[];
}
```

There doesn't seem to be any `exclude` field there :/

**Comment by undeadcat** [comment]
This was also requested by users of the TSLint integration in WebStorm/Intellij IDEA  in [this issue](https://youtrack.jetbrains.com/issue/WEB-33060).

From looking at the code, it seems like linterOptions.exclude is only handled in 'runner.ts' [here](https://github.com/palantir/tslint/blob/6b52df3fddfef577e3f7703852642553107aec01/src/runner.ts#L287) which seems to be the command-line implementation.

**Comment by saadq** [comment]
Thanks for the find. I've been just spawning a `tslint` child process programatically instead of using the TSLint Node API due to this :/

**Comment by biolauri** [comment]
I assume, this one is already fixed but not shipped/released. Would anyone be so kind to give a rough estimation when we'll find this via npm?  
Just to plan a bit if it's worth changing our config or better to wait.

And, of course: Thank you very much for the fix! :)

**Comment by JoshuaKGoldberg** [comment]
Fixed by #4143. 🎉

No estimate for now on an npm package release date, sorry!
